src/Engine/Rendering/Shaders/IShader.hpp

```diff
@@ -91,8 +91,9 @@ namespace Engine
 				}
 
 			public:
-				virtual bool VertexShader(Vertex& v0, Vertex& v1, Vertex& v2, const MVPTransform& mvp_mats)									  = 0;
-				virtual void FragmentShader(HSVColor& out_color, const Triangle& triangle, float barcoord0, float barcoord1, float barcoord2) = 0;
+				virtual bool VertexShader(Vertex& v0, Vertex& v1, Vertex& v2, const MVPTransform& mvp_mats) = 0;
+				virtual HSVColor FragmentShader(
```

Comment:
Food for thought, perhaps the HSV color conversion can be deferred? Such as another model obscuring it.

Reply:
Yeah I agree, however I don't think it's need right now, because if everything goes well I can go away with the HSVColor entirely, I just need to some how figure out the 3 dimensional RGB mapping -> output attributes

If that turns out to be too hard/not desirable i guess extracting this would make sense, previously it was the different renderers (Dithered/ASCII/Greyscale etc..) job to do this, but I did away with that idea as IMO it was also not the renderers concern 

If I were to extract this to a separate component i'd possibly connect that component to each renderer independently, what do you think?

Best case scenario is not having HSV at all though